### PR TITLE
Fix scipy constraint and add pip check

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

A downstream package I work on is failing `pip check` for python 3.9 (and would be for python 3.8 if I were checking it) because of the missing constraint on `scipy` for python <= 3.9.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -40,7 +40,8 @@ requirements:
   run:
     - python
     - numpy
-    - scipy >=1.8
+    - scipy >=1.8,<1.9.2  # [py <= 39]
+    - scipy >=1.8  # [py > 39]
```
Comment (xylar) [inline recipe/meta.yaml:44]: See https://github.com/scikit-image/scikit-image/blob/v0.20.0/pyproject.toml#L40-L41

**Review by stefanv — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (stefanv) [reply recipe/meta.yaml:44]: That constraint was probably a mistake; we're getting rid of it in the next version, being released this week.

**Comment by xylar** [comment]
@stefanv, an alternative would be to patch the `setup.py` in this build so it doesn't have the separate constraints for python <=3.9.  Let me know if you prefer that.  Obviously, in a week this will be a moot point.

**Comment by xylar** [comment]
I think it's better not to do the patch option.  Would someone from @conda-forge/scikit-image be willing to merge this?

File: recipe/meta.yaml (hunk 1 above)
Comment (valeriupredoi) [reply recipe/meta.yaml:44]: ah good spotting that! I have been scratching me head for the past hour why our Python 3.8 and 3.9 tests die with a fairly obtuse`ContextualVersionConflict` (see [fails](https://github.com/ESMValGroup/ESMValTool/actions/runs/4677958708/jobs/8286095951) ) :grin:

**Comment by xylar** [comment]
@conda-forge/scikit-image, again, would you be willing to review and merge?  This is causing trouble for others, too, it seems.

**Comment by valeriupredoi** [comment]
@conda-forge/scikit-image guys, why are there multiple builds (with different hashes) for the same architecture pushed 4 days and some 19 hours ago?

- [linux-64/scikit-image-0.20.0-py39ha2d197d_1.conda](https://anaconda.org/conda-forge/scikit-image/0.20.0/download/linux-64/scikit-image-0.20.0-py39ha2d197d_1.conda)
- [linux-64/scikit-image-0.20.0-py39h7c7b5a8_1.conda](https://anaconda.org/conda-forge/scikit-image/0.20.0/download/linux-64/scikit-image-0.20.0-py39h7c7b5a8_1.conda)

The latter one (which is slightly younger than the top one) is still making our tests fail with the odd scipy constraint, see the Install [step here](https://github.com/ESMValGroup/ESMValTool/actions/runs/4717367563/jobs/8365918811)

**Comment by xylar** [comment]
@valeriupredoi, I can answer the first question but I can't tell you why your CI is failing.

`linux-64/scikit-image-0.20.0-py39ha2d197d_1.conda` is for [PyPy](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=688689&view=logs&jobId=517fe804-fa30-5dc2-1413-330699242c05&j=9a864fd9-6c8f-52ca-79ce-2aa6dca1a1de&t=10fc5aa2-324e-5982-4c88-6b31fcab16b3)

`linux-64/scikit-image-0.20.0-py39h7c7b5a8_1.conda` is for [CPython](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=688689&view=logs&jobId=517fe804-fa30-5dc2-1413-330699242c05&j=bb1c2637-64c6-57bd-9ea6-93823b2df951&t=350df31b-3291-5209-0bb7-031395f0baa1)

It make sense you're getting the latter because you're building with CPython.

**Comment by xylar** [comment]
@valeriupredoi, your CI is failing because it's still picking up build 0:
```
  + scikit-image                     0.20.0  py39h7c7b5a8_0          conda-forge/linux-64      10MB
```
We will need to do a repopatch to fix that.

**Comment by valeriupredoi** [comment]
ah snap! You're totally right @xylar - cross-eyed myself and saw the `_1` from the next package under it, indeed it's picking up `py39h7c7b5a8_0` - what you recommend we do - wait for you guys to repopatch or release 0.21.x or we pin `scikit-image` in our package and rebuild - I'd be weary to do the latter since then we'd have to rebuild again when this is fixed upstream :beer:

**Comment by xylar** [comment]
Let me put in a repopatch.  It's annoying but not that hard.

**Comment by valeriupredoi** [comment]
awesome, cheers muchly, mate! :beer:

**Comment by xylar** [comment]
@valeriupredoi, I'll drag you down, too, by making you review ;-)

**Comment by valeriupredoi** [comment]
heh, no probs, mate - shall dos!

**Comment by xylar** [comment]
Please have a look at https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/436

**Pull request merged**